commands/cli_nukeallresources_test.go

```diff
@@ -0,0 +1,547 @@
+
+	if collections.ListContains(args.postNukeResources, "ec2") {
+		require.Truef(
+			t, ec2Exists,
+			"Test EC2 instance deleted when it should not have - name - %s - instanceID - %s",
+			ec2InstanceName, resourceTypeIdentifierMap["ec2"],
+		)
+		logging.Logger.Debug("Expected: Post nuke check - ec2 exists")
+	} else {
+		require.Falsef(
+			t, ec2Exists,
+			"Test EC2 instance not deleted - name - %s - instanceId - %s",
+			ec2InstanceName, resourceTypeIdentifierMap["ec2"],
+		)
+		logging.Logger.Debug("Expected: Post nuke check - ec2 deleted")
+	}
+	if collections.ListContains(args.postNukeResources, "s3") {
+		require.Truef(t, s3Exists, "Test S3 bucket deleted when it should not have - %s", bucketName)
+		logging.Logger.Debug("Expected: Post nuke check - s3 exists")
+	} else {
+		require.Falsef(t, s3Exists, "Test S3 bucket not deleted - %s", bucketName)
+		logging.Logger.Debug("Expected: Post nuke check - s3 deleted")
+	}
+}
+
+// testNukeAllResourcesArgs bundles up testNukeAllResources args
+type testNukeAllResourcesArgs struct {
+	awsSession              *session.Session // main AWS session which will be used to create/delete test resources
+	assumeRoleArn           string           // ARN of role to assume
+	ignoreErrResourceTypes  []string         // list of resource types to ignore AWS errors
+	getResourcesShouldFail  bool             // whether GetAllResources should return err
+	getResourcesOutputTypes []string         // what resourcetypes from test resources should GetAllResources return
+	nukeResourcesShouldFail bool             // flag which decides whether NukeAllResources should return err
+	postNukeResources       []string         // what resourcetypes from test resources should exist after NukeAllResources
+}
+
+// getIAMRolePolicyMamp maps a role type to a corresponding policy
+func getTestRolePolicyMap() map[string]string {
+	return map[string]string{
+		"NoPerms": aws.TrimPolicyDocument(`{
+			"Version": "2012-10-17",
+			"Statement": {
+				"Effect": "Allow",
+				"Action": "ec2:DescribeRegions",
+				"Resource": "*"
+			}
+		}`),
+		"ROPerms": aws.TrimPolicyDocument(`{
+			"Version": "2012-10-17",
+			"Statement": {
+				"Effect": "Allow",
+				"Action": [
+					"ec2:Describe*",
+					"s3:List*",
+					"s3:Get*"
+				],
+				"Resource": "*"
+			}
+		}`),
+		"EC2ROS3RWPerms": aws.TrimPolicyDocument(`{
+			"Version": "2012-10-17",
+			"Statement": {
+				"Effect": "Allow",
+				"Action": [
+					"ec2:Describe*",
+					"s3:*"
+				],
+				"Resource": "*"
+			}
+		}`),
+	}
+}
+
+// TestNukeAllResourcesNoPerms tests deletion of test S3 and EC2 instance with an assumed
+// role which does not have S3/EC2 read/write permissions
+func TestNukeAllResources(t *testing.T) {
+	t.Parallel()
+	// Create a top level AWS session object which will be used to create/destroy roles
+	// and resources. Specifically use us-east-1 as we are running >= 8 tests and vCpu limits
```

Comment:
Are the default limits for `us-east-1` any higher? Or is it just that you requested limited increases already for that region?

Reply:
These are the default limits. I did not request an increase.